### PR TITLE
Suggested changes to linux_config to make more user customizable.

### PR DESCRIPTION
**Pull request by savagesmc**
I have done a few cleanups of things i saw in the vimrc file and the other config files, as well as moving user specific configuration (user name in .gitconfig for example) out to user specific files that can be provided in the root directory through another mechanism.

### CODE REVIEW

File: gitconfig [hunk 1]
```diff
@@ -1,8 +1,5 @@
-[user]
-	name = McCabe, Robert J
-	email = Robert.McCabe@rockwellcollins.com
 [sendemail]
-    smtpserver = notessmtp.rockwellcollins.com
+    smtpserver = smtpimr.rockwellcollins.com
```
Comment (rjmccabe3701) [inline gitconfig]: I dont agree with this.

File: vimrc [hunk 2]
```diff
@@ -33,12 +29,20 @@ Plugin 'shougo/neocomplete.vim'
 Plugin 'tpope/vim-markdown'
 Plugin 'chazy/cscope_maps'
 Plugin 'nvie/vim-flake8'
+Plugin 'vim-scripts/a.vim'
```
Comment (rjmccabe3701) [inline vimrc:32]: What is a.vim?

File: gitconfig (hunk 1 above)
Comment (rjmccabe3701) [inline gitconfig]: I dont agree with this.

File: vimrc [hunk 3]
```diff
@@ -58,6 +62,7 @@ filetype plugin indent on    " required
 " https://github.com/kien/ctrlp.vim/blob/master/doc/ctrlp.txt
 let g:ctrlp_max_files = 0
 let g:ctrlp_max_depth = 40
+set wildignore+=.git\*,.hg\*,.svn\*,*.d,*.dbo,*.o,*.ti,*.a,*.so
```
Comment (rjmccabe3701) [inline vimrc:65]: what does this ignore?

**Review by rjmccabe3701 — CHANGES_REQUESTED** [review]

File: vimrc (hunk 2 above)
Comment (savagesmc) [reply vimrc:32]: Alternate macros to switch between header and source files easily.

File: vimrc (hunk 3 above)
Comment (savagesmc) [reply vimrc:65]: Skips over repo overhead folders and build intermediate files for ctrl-p.

**Pull request merged**